src/components/Editor/EditorControlPane/Widget.js

```diff
@@ -104,7 +104,9 @@ export default class Widget extends Component {
 
   validatePattern(field, value) {
     const pattern = field.get('pattern', false);
-    if (pattern && !RegExp(pattern.first()).test(value)) {
+    const hasValue = typeof value === 'string' && value.length > 0;
```

Comment:
This would stop non-string fields (e.g. numbers) from being pattern validated.

Instead of adding logic, you could split out the empty field test logic from `validatePresence` and reuse it here.